homeassistant/components/bmw_connected_drive/select.py

```diff
@@ -0,0 +1,139 @@
+"""Select platform for BMW."""
+from collections.abc import Callable, Coroutine
+from dataclasses import dataclass
+import logging
+from typing import Any
+
+from bimmer_connected.vehicle import MyBMWVehicle
+from bimmer_connected.vehicle.charging_profile import ChargingMode
+
+from homeassistant.components.select import SelectEntity, SelectEntityDescription
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import PERCENTAGE, UnitOfElectricCurrent
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import BMWBaseEntity
+from .const import DOMAIN
+from .coordinator import BMWDataUpdateCoordinator
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@dataclass
+class BMWRequiredKeysMixin:
+    """Mixin for required keys."""
+
+    current_option: Callable[[MyBMWVehicle], str]
+    remote_service: Callable[[MyBMWVehicle, str], Coroutine[Any, Any, Any]]
+
+
+@dataclass
+class BMWSelectEntityDescription(SelectEntityDescription, BMWRequiredKeysMixin):
+    """Describes BMW sensor entity."""
+
+    is_available: Callable[[MyBMWVehicle], bool] = lambda _: False
+    dynamic_options: Callable[[MyBMWVehicle], list[str]] | None = None
+
+
+SELECT_TYPES: dict[str, BMWSelectEntityDescription] = {
+    # --- Generic ---
+    "target_soc": BMWSelectEntityDescription(
+        key="target_soc",
+        name="Target SoC",
+        is_available=lambda v: v.is_remote_set_target_soc_enabled,
+        options=[str(i * 5 + 20) for i in range(17)],
```

Comment:
Why is this not represented by a number entity with min 20, max 100, stepsize 1?

Reply:
We need step size 5 as the API doesn't accept anything in between.
I wasn't able to do that with number steps somehow.

Reply:
Did you try setting `native_step` to 5?